queue/retry_test.go

```diff
@@ -285,14 +285,6 @@ func TestRetryHandlerImplementations(t *testing.T) {
 
 					assert.Error(t, rh.Put(ctx, nil))
 				},
-				"PutIsIdempotent": func(ctx context.Context, t *testing.T, makeQueueAndRetryHandler func(opts amboy.RetryHandlerOptions) (*mockRemoteQueue, amboy.RetryHandler, error)) {
```

Comment:
It's not particularly important. It was just a small optimization for a single retry handler on a single host to avoid duplicates (but would not avoid duplicates across app servers). Since most jobs funnel into `Put()` from the execution of jobs, the likelihood of in-memory duplicates is really low. The only way there could be in-memory duplicates would be from the "monitor stale retrying jobs" goroutine, which is only for exceptional cases.